src/resolver.js

```diff
@@ -1,89 +1,73 @@
 'use strict'
 
 const mh = require('multihashes')
-const promisify = require('promisify-es6')
 const CID = require('cids')
 const debug = require('debug')
-const tryEach = require('async/tryEach')
-const waterfall = require('async/waterfall')
 const log = debug('jsipfs:http:response:resolver')
 log.error = debug('jsipfs:http:response:resolver:error')
 const dirView = require('./dir-view')
+const pTryEeach = require('./utils/p-try-each')
 
 const INDEX_HTML_FILES = [
   'index.html',
   'index.htm',
   'index.shtml'
 ]
 
-const findIndexFile = (ipfs, path, callback) => {
-  return tryEach(INDEX_HTML_FILES.map(file => {
-    return (cb) => {
-      waterfall([
-        (cb) => ipfs.files.stat(`${path}/${file}`, cb),
-        (stats, cb) => cb(null, {
-          name: file,
-          cid: new CID(stats.hash)
-        })
-      ], cb)
+const findIndexFile = (ipfs, path) => {
+  return pTryEeach(INDEX_HTML_FILES.map(file => {
+    return async () => {
+      const stats = await ipfs.files.stat(`${path}/${file}`)
+
+      return {
+        name: file,
+        cid: new CID(stats.hash)
+      }
     }
-  }), callback)
+  }))
 }
 
-const directory = promisify((ipfs, path, cid, callback) => {
-  // Test if it is a Website
-  findIndexFile(ipfs, path, (err, res) => {
-    if (err) {
-      if (err.message.includes('does not exist')) {
-        // not a website, just show a directory listing
-        return ipfs.dag.get(cid, (err, result) => {
-          if (err) {
-            return callback(err)
-          }
-
-          return callback(null, dirView.render(path, result.value.Links))
-        })
-      }
+const directory = async (ipfs, path, cid) => {
+  try {
+    const res = await findIndexFile(ipfs, path)
```

Comment:
Can you put the comment back "test if it is a website"? I feel it is important here